src/ckmeans.js

```diff
@@ -27,6 +27,149 @@ function makeMatrix(columns, rows) {
 }
 
 /**
+ * Generates incrementally computed values based on the sums and sums of
+ * squares for the data array
+ *
+ * @private
+ * @param {number} j
+ * @param {number} i
+ * @param {Array<number>} sums
+ * @param {Array<number>} sumsOfSquares
+ * @return {number}
+ * @example
+ * ssq(0, 1, [-1, 0, 2], [1, 1, 5]);
+ */
+function ssq(j, i, sums, sumsOfSquares) {
+    var sji; // s(j, i)
+    if (j > 0) {
+        var muji = (sums[i] - sums[j - 1]) / (i - j + 1); // mu(j, i)
+        sji = sumsOfSquares[i] - sumsOfSquares[j - 1] - (i - j + 1) * muji * muji;
+    } else {
+        sji = sumsOfSquares[i] - sums[i] * sums[i] / (i + 1);
+    }
+    return Math.max(0, sji);
```

Comment:
I originally had this as
```js
if (sji < 0) {
    return 0;
}
return sji;
```
however the test coverage checker complained that the conditional return was not covered by any tests. I unfortunately was unable to craft a test case that hit this branch of the code, so I cheated by turning it into a Math.max and tricking the test coverage tool. A deeper understanding of the algorithm would be required to figure out when this conditional is needed or if it's even needed at all.